controller/app/controllers/deployments_controller.rb

```diff
@@ -40,6 +40,9 @@ def create
     return render_error(:unprocessable_entity, "Cannot specify a git deployment and a binary artifact deployment",
                           -1, "artifact_url") if artifact_url && ref
 
+    return render_error(:unprocessable_entity, "Must specify a git deployment or a binary artifact deployment",
+                          -1, "artifact_url") unless artifact_url || ref
```

Comment:
You pass `"artifact_url"` as the field in the `render_error`. This is the field that the error supposedly applies to. Since it applies to `artifact_url` and `ref`, I would probably leave it blank.